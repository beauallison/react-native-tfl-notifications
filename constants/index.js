module.exports = {
  MODES: {
    TUBE: 'tube',
  },
  LINES: {
    TUBE: {
      BAKERLOO: 'bakerloo',
      CENTRAL: 'central',
      DISTRICT: 'district',
      CIRCLE: 'circle',
      HAMMERSMITH: 'hammersmith-city',
      JUBILEE: 'jubilee',
      METROPOLITAN: 'metropolitan',
      NORTHERN: 'northern',
      PICCADILLY: 'piccadilly',
      VICTORIA: 'victoria',
      WATERLOO: 'waterloo-city',
    },
  },
};

