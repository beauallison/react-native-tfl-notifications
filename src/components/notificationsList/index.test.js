import _ from 'lodash';
import React from 'react';
import { shallow } from 'enzyme';
import NotificationsList from './index';
import { LINES } from '../../../constants';

describe('components/notificationsList', () => {
  const data = _.map(LINES, (station, ID) => ({
    ID,
    station,
    onPress: (() => Promise.resolve(true)),
  }));

  it('should render notifications List', () => {
    const wrapper = shallow(<NotificationsList data={data} save={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
