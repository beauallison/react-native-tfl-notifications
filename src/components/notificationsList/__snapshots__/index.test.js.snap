// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/notificationsList should render notifications List 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationsList
    data={
        Array [
            Object {
              "ID": "BAKERLOO",
              "onPress": [Function],
              "station": "bakerloo",
            },
            Object {
              "ID": "CENTRAL",
              "onPress": [Function],
              "station": "central",
            },
            Object {
              "ID": "DISTRICT",
              "onPress": [Function],
              "station": "district",
            },
            Object {
              "ID": "CIRCLE",
              "onPress": [Function],
              "station": "circle",
            },
            Object {
              "ID": "HAMMERSMITH",
              "onPress": [Function],
              "station": "hammersmith-city",
            },
            Object {
              "ID": "JUBILEE",
              "onPress": [Function],
              "station": "jubilee",
            },
            Object {
              "ID": "LONDON",
              "onPress": [Function],
              "station": "london-overground",
            },
            Object {
              "ID": "METROPOLITAN",
              "onPress": [Function],
              "station": "metropolitan",
            },
            Object {
              "ID": "NORTHERN",
              "onPress": [Function],
              "station": "northern",
            },
            Object {
              "ID": "PICCADILLY",
              "onPress": [Function],
              "station": "piccadilly",
            },
            Object {
              "ID": "VICTORIA",
              "onPress": [Function],
              "station": "victoria",
            },
            Object {
              "ID": "WATERLOO",
              "onPress": [Function],
              "station": "waterloo-city",
            },
          ]
    }
    save={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FlatList
          data={
                    Array [
                              Object {
                                "ID": "BAKERLOO",
                                "onPress": [Function],
                                "station": "bakerloo",
                              },
                              Object {
                                "ID": "CENTRAL",
                                "onPress": [Function],
                                "station": "central",
                              },
                              Object {
                                "ID": "DISTRICT",
                                "onPress": [Function],
                                "station": "district",
                              },
                              Object {
                                "ID": "CIRCLE",
                                "onPress": [Function],
                                "station": "circle",
                              },
                              Object {
                                "ID": "HAMMERSMITH",
                                "onPress": [Function],
                                "station": "hammersmith-city",
                              },
                              Object {
                                "ID": "JUBILEE",
                                "onPress": [Function],
                                "station": "jubilee",
                              },
                              Object {
                                "ID": "LONDON",
                                "onPress": [Function],
                                "station": "london-overground",
                              },
                              Object {
                                "ID": "METROPOLITAN",
                                "onPress": [Function],
                                "station": "metropolitan",
                              },
                              Object {
                                "ID": "NORTHERN",
                                "onPress": [Function],
                                "station": "northern",
                              },
                              Object {
                                "ID": "PICCADILLY",
                                "onPress": [Function],
                                "station": "piccadilly",
                              },
                              Object {
                                "ID": "VICTORIA",
                                "onPress": [Function],
                                "station": "victoria",
                              },
                              Object {
                                "ID": "WATERLOO",
                                "onPress": [Function],
                                "station": "waterloo-city",
                              },
                            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
/>,
        <SaveButton
          onPress={[Function]}
/>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "ID": "BAKERLOO",
              "onPress": [Function],
              "station": "bakerloo",
            },
            Object {
              "ID": "CENTRAL",
              "onPress": [Function],
              "station": "central",
            },
            Object {
              "ID": "DISTRICT",
              "onPress": [Function],
              "station": "district",
            },
            Object {
              "ID": "CIRCLE",
              "onPress": [Function],
              "station": "circle",
            },
            Object {
              "ID": "HAMMERSMITH",
              "onPress": [Function],
              "station": "hammersmith-city",
            },
            Object {
              "ID": "JUBILEE",
              "onPress": [Function],
              "station": "jubilee",
            },
            Object {
              "ID": "LONDON",
              "onPress": [Function],
              "station": "london-overground",
            },
            Object {
              "ID": "METROPOLITAN",
              "onPress": [Function],
              "station": "metropolitan",
            },
            Object {
              "ID": "NORTHERN",
              "onPress": [Function],
              "station": "northern",
            },
            Object {
              "ID": "PICCADILLY",
              "onPress": [Function],
              "station": "piccadilly",
            },
            Object {
              "ID": "VICTORIA",
              "onPress": [Function],
              "station": "victoria",
            },
            Object {
              "ID": "WATERLOO",
              "onPress": [Function],
              "station": "waterloo-city",
            },
          ],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FlatList
            data={
                        Array [
                                    Object {
                                      "ID": "BAKERLOO",
                                      "onPress": [Function],
                                      "station": "bakerloo",
                                    },
                                    Object {
                                      "ID": "CENTRAL",
                                      "onPress": [Function],
                                      "station": "central",
                                    },
                                    Object {
                                      "ID": "DISTRICT",
                                      "onPress": [Function],
                                      "station": "district",
                                    },
                                    Object {
                                      "ID": "CIRCLE",
                                      "onPress": [Function],
                                      "station": "circle",
                                    },
                                    Object {
                                      "ID": "HAMMERSMITH",
                                      "onPress": [Function],
                                      "station": "hammersmith-city",
                                    },
                                    Object {
                                      "ID": "JUBILEE",
                                      "onPress": [Function],
                                      "station": "jubilee",
                                    },
                                    Object {
                                      "ID": "LONDON",
                                      "onPress": [Function],
                                      "station": "london-overground",
                                    },
                                    Object {
                                      "ID": "METROPOLITAN",
                                      "onPress": [Function],
                                      "station": "metropolitan",
                                    },
                                    Object {
                                      "ID": "NORTHERN",
                                      "onPress": [Function],
                                      "station": "northern",
                                    },
                                    Object {
                                      "ID": "PICCADILLY",
                                      "onPress": [Function],
                                      "station": "piccadilly",
                                    },
                                    Object {
                                      "ID": "VICTORIA",
                                      "onPress": [Function],
                                      "station": "victoria",
                                    },
                                    Object {
                                      "ID": "WATERLOO",
                                      "onPress": [Function],
                                      "station": "waterloo-city",
                                    },
                                  ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
/>,
          <SaveButton
            onPress={[Function]}
/>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "ID": "BAKERLOO",
                "onPress": [Function],
                "station": "bakerloo",
              },
              Object {
                "ID": "CENTRAL",
                "onPress": [Function],
                "station": "central",
              },
              Object {
                "ID": "DISTRICT",
                "onPress": [Function],
                "station": "district",
              },
              Object {
                "ID": "CIRCLE",
                "onPress": [Function],
                "station": "circle",
              },
              Object {
                "ID": "HAMMERSMITH",
                "onPress": [Function],
                "station": "hammersmith-city",
              },
              Object {
                "ID": "JUBILEE",
                "onPress": [Function],
                "station": "jubilee",
              },
              Object {
                "ID": "LONDON",
                "onPress": [Function],
                "station": "london-overground",
              },
              Object {
                "ID": "METROPOLITAN",
                "onPress": [Function],
                "station": "metropolitan",
              },
              Object {
                "ID": "NORTHERN",
                "onPress": [Function],
                "station": "northern",
              },
              Object {
                "ID": "PICCADILLY",
                "onPress": [Function],
                "station": "piccadilly",
              },
              Object {
                "ID": "VICTORIA",
                "onPress": [Function],
                "station": "victoria",
              },
              Object {
                "ID": "WATERLOO",
                "onPress": [Function],
                "station": "waterloo-city",
              },
            ],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
